### PR TITLE
Fixing bug in building for GPU for 0.12.1

### PR DESCRIPTION
**Pull request by nfwvogt**

### CODE REVIEW

File: roqoqo-quest/src/quest_bindings.rs [hunk 1]
```diff
@@ -74,21 +75,133 @@ impl Qureg {
...
+        let mut density_matrix_flattened_row_major: Vec<Complex64> =
+            Vec::with_capacity(4_usize.pow(self.number_qubits()));
+        if self.is_density_matrix {
+            for row in 0..dimension {
+                for column in 0..dimension {
+                    let density_index = row + column * dimension;
+                    unsafe {
+                        let real_amp = *self
+                            .quest_qureg
+                            .stateVec
```
Comment (kbarkhqs) [inline roqoqo-quest/src/quest_bindings.rs:158]: Same here, should this be stateVec?

File: roqoqo-quest/src/quest_bindings.rs [hunk 2]
```diff
@@ -74,21 +75,133 @@ impl Qureg {
...
+                        let imag_amp_column = *self
+                            .quest_qureg
+                            .stateVec
+                            .imag
+                            .wrapping_add(column.try_into().expect("Indexing error"));
+
+                        Complex64::new(real_amp_row, imag_amp_row)
+                            * Complex64::new(real_amp_column, imag_amp_column).conj()
+                    };
+                    density_matrix_flattened_row_major.push(value);
```
Comment (kbarkhqs) [inline roqoqo-quest/src/quest_bindings.rs:199]: How come this doesn't throw an error, given that we don't have a density matrix qureg but call `get_density_matrix`?

File: roqoqo-quest/src/quest_bindings.rs [hunk 3]
```diff
@@ -74,21 +75,133 @@ impl Qureg {
...
         if self.is_density_matrix {
             for index in 0..dimension {
+                let density_index = index + index * dimension;
                 unsafe {
-                    probabilites.push(
-                        quest_sys::getDensityAmp(self.quest_qureg, index.into(), index.into()).real,
-                    )
+                    let real_amp = *self
+                        .quest_qureg
+                        .stateVec
```
[the remark this answers is not in the document]
Comment (nfwvogt) [reply roqoqo-quest/src/quest_bindings.rs:87]: Yes internally everything is stored as a statevec if you compare the QuEST code

File: roqoqo-quest/src/quest_bindings.rs (hunk 1 above)
Comment (nfwvogt) [reply roqoqo-quest/src/quest_bindings.rs:158]: Same as above

File: roqoqo-quest/src/quest_bindings.rs (hunk 2 above)
Comment (nfwvogt) [reply roqoqo-quest/src/quest_bindings.rs:199]: Every state vector can always be represented as the density matrix of a pure state, so it makes sense to have this.

**Review by kbarkhqs — APPROVED** [review]

**Pull request merged**